src/bos/operators/base.py

```diff
@@ -162,6 +162,7 @@ def _update_database(self, components: List[dict], additional_fields: dict=None)
         for component in components:
             patch = {
                 'id': component['id'],
+                'enabled': component['enabled'], # Enabled state overwrites previous state
```

Comment:
I think it would make more sense to remove this and have "if not component['enabled']: patch['enabled'] = component['enabled'].   The reason is that, unless we've decided to disable a node, this is essentially cached from the last time we queried from this nodes data.  That means if someone disables a node while an operator is in the middle of processing a group of nodes containing that node, the operator would re-apply the cached "enabled" state and potentially flip the component back to enabled.

Reply:
I agree with that logic. I will make the change and re-test.